packages/auth/src/api/index.ts

```diff
@@ -181,7 +181,12 @@ export async function _performFetchWithErrorHandling<V>(
     if (e instanceof FirebaseError) {
       throw e;
     }
-    _fail(auth, AuthErrorCode.NETWORK_REQUEST_FAILED);
+    if (e instanceof Error) {
```

Comment:
I am not sure when we would actually end up here, because all the paths in the try block seem to be throwing FirebaseError. I think it is fine to make this change, so we don't mask any non-networking errors with NETWORK_REQUEST_FAILED.

Reply:
Ack. Removed the if branch and do `_fail(auth, AuthErrorCode.INTERNAL_ERROR, {"message": String(e)});` instead as suggested in the other comment because this should be sufficient.